src/App.tsx

```diff
@@ -9,39 +9,31 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { loadPosts, postsSlice } from './features/posts/postsSlice';
+import { selectedPostSlice } from './features/selectedPost/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const {
+    posts,
+    loaded: loaded,
```

Comment:
The `loaded` property is being renamed to `loaded` unnecessarily. You can simply use `loaded` without renaming it, as it doesn't change the property name.